apps/trading/lib/hooks/use-vega-transaction-toasts.tsx

```diff
@@ -0,0 +1,572 @@
+        </Details>
+      );
+    }
+  }
+
+  if (isBatchMarketInstructionsTransaction(tx.body)) {
+    return <Details>{t('Batch market instruction')}</Details>;
+  }
+
+  return null;
+};
+
+type VegaTxToastContentProps = { tx: VegaStoredTxState };
+
+const VegaTxRequestedToastContent = ({ tx }: VegaTxToastContentProps) => (
+  <div>
+    <h3 className="font-bold">{t('Action required')}</h3>
+    <p>
+      {t(
+        'Please go to your Vega wallet application and approve or reject the transaction.'
+      )}
+    </p>
+    <VegaTransactionDetails tx={tx} />
+  </div>
+);
+
+const VegaTxPendingToastContentProps = ({ tx }: VegaTxToastContentProps) => {
+  const explorerLink = useLinks(DApp.Explorer);
+  return (
+    <div>
+      <h3 className="font-bold">{t('Awaiting confirmation')}</h3>
+      <p>{t('Please wait for your transaction to be confirmed')}</p>
+      {tx.txHash && (
+        <p className="break-all">
+          <ExternalLink
+            href={explorerLink(
+              EXPLORER_TX.replace(':hash', prepend0x(tx.txHash))
+            )}
+            rel="noreferrer"
+          >
+            {t('View in block explorer')}
+          </ExternalLink>
+        </p>
+      )}
+      <VegaTransactionDetails tx={tx} />
+    </div>
+  );
+};
+
+const VegaTxCompleteToastsContent = ({ tx }: VegaTxToastContentProps) => {
+  const { createEthWithdrawalApproval } = useEthWithdrawApprovalsStore(
+    (state) => ({
+      createEthWithdrawalApproval: state.create,
+    })
+  );
+  const explorerLink = useLinks(DApp.Explorer);
+  if (isWithdrawTransaction(tx.body)) {
+    const completeWithdrawalButton = tx.withdrawal && (
+      <div className="mt-[10px]">
+        <Button
+          size="xs"
+          onClick={() => {
+            createEthWithdrawalApproval(
+              tx.withdrawal as WithdrawalBusEventFieldsFragment,
+              tx.withdrawalApproval
+            );
+          }}
+        >
+          {t('Complete withdrawal')}
+        </Button>
+      </div>
+    );
+    return (
+      <div>
+        <h3 className="font-bold">{t('Funds unlocked')}</h3>
+        <p>{t('Your funds have been unlocked for withdrawal')}</p>
+        {tx.txHash && (
+          <p className="break-all">
+            <ExternalLink
+              href={explorerLink(
+                EXPLORER_TX.replace(':hash', prepend0x(tx.txHash))
+              )}
+              rel="noreferrer"
+            >
+              {t('View in block explorer')}
+            </ExternalLink>
+          </p>
+        )}
+        <VegaTransactionDetails tx={tx} />
+        {completeWithdrawalButton}
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <h3 className="font-bold">{t('Confirmed')}</h3>
+      <p>{t('Your transaction has been confirmed ')}</p>
+      {tx.txHash && (
+        <p className="break-all">
+          <ExternalLink
+            href={explorerLink(
+              EXPLORER_TX.replace(':hash', prepend0x(tx.txHash))
```

Comment:
suggestion (non blocking): For other tx links we aren't prepending 0x can we make this consistent? After checking with @edd block explorer will work both ways but as core doesn't need it we should be consistent